source/appModules/wwahost.py

```diff
@@ -1,14 +1,39 @@
 #A part of NonVisual Desktop Access (NVDA)
 #This file is covered by the GNU General Public License.
 #See the file COPYING for more details.
-#Copyright (C) 2012 NV Access Limited
+#Copyright (C) 2012-2019 NV Access Limited, Joseph Lee
+
+"""App module host for Windows 8.x and 10 web apps hosted by wwahost.exe.
+In Windows 8, apps written in Javascript are executed inside WWAHost, including some WinRT apps.
+In Windows 10, progressive web apps (PWA) and friends are hosted inside this process.
+App modules wishing to support apps hosted inside this process must subclass the AppModule class.
+"""
 
 from comtypes import COMError
+import ctypes
 import IAccessibleHandler
 from NVDAObjects.IAccessible.MSHTML import Body
 import appModuleHandler
 import controlTypes
 import winUser
+import winKernel
+
+def getAppNameFromHost(processId):
+	# Some apps that come with Windows 8 and 8.1 are hosted by wwahost.exe.
+	# App modules for these are named after the hosted app name.
+	processHandle = winKernel.openProcess(winKernel.SYNCHRONIZE|winKernel.PROCESS_QUERY_INFORMATION,False,processId)
+	length = ctypes.c_uint()
+	buf = winKernel.kernel32.GetApplicationUserModelId(processHandle, ctypes.byref(length), None)
+	appModel = ctypes.create_unicode_buffer(buf)

```

Comment:
GetApplicationUserModelId returns an error code, not a length. You should create a buffer using length.value. You can ignore the error code in this case. In any case, I don't recommend using the name buf to store a length; it's confusing.